plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -423,6 +463,94 @@ func (s *SQLServer) Init() error {
 	return nil
 }
 
+// Get Token Provider by loading cached token or refreshed token
+func (s *SQLServer) getTokenProvider() (func() (string, error), error) {
```

Comment:
Can we move token related methods to a different class? It would keep the main class short and simple

Reply:
I feel the methods are not reusable as such to warrant a refactor to another class. And after moving the cache variable and it's lock into the SQLServer struct, those methods are more bound and separating the methods to another class won't add value. I'd prefer to keep them here for now unless we change the caching mechanism to a more reusable structure.